src/core/server/ui_settings/create_or_upgrade_saved_config/create_or_upgrade_saved_config.ts

```diff
@@ -41,8 +42,15 @@ export async function createOrUpgradeSavedConfig(
 
   // default to the attributes of the upgradeableConfig if available
   const attributes = defaults(
-    { buildNum },
-    upgradeableConfig ? (upgradeableConfig.attributes as any) : {}
+    {
+      buildNum,
+      // If there is an upgradeableConfig, attempt to transform the defaultIndex attribute to ensure it is set properly in 8.3.1+
+      ...(await transformDefaultIndex({
+        savedObjectsClient,
+        configAttributes: upgradeableConfig?.attributes,
+      })),
+    },
+    upgradeableConfig?.attributes
```

Comment:
optional: I was thinking that since we are adding transforms to our dev docs as a mechanism that can be used to fix bugs like this, it might be nice to isolate this further so that it is dead simple if another one needs to be added. For example, importing an array of transform Fns here that are maintained in a separate file, so that we can use TS to enforce that each transform fn has the right shape:

```ts
// saved_objects/transforms.ts
type TransformConfigFn = (Params) => Promise<ReturnType>;
const fn1 = (...) => ({...});
const fn2 = (...) => ({...});
export const transforms: TransformConfigFn[] = [fn1, fn2];

// create_or_upgrade_saved_config.ts
transforms.forEach(async (fn) => await fn({ savedObjectsClient, configAttributes }));
```

Then if someone ever does need to use this in the future (hopefully they don't), all that's required is adding a new transform to the file + a unit test. Similar to what we've done with migrations.

Reply:
Done in a33ef0598cdd8ff7b15f34dc7f32f0cdf0cc0486!